### PR TITLE
feat(dropdown): Added dropdownMenu directive to use for context menus

### PR DESCRIPTION
**Pull request by rasmusvhansen**
I have added a feature to dropdown, so it can be used as a context menu. Meaning it will show up at the specified coordinates. See demo.js and demo.html for details.

It is implemented as a dropdownMenu directive. So elements marked with dropdown-menu will register with the dropdown controller, and when dropdown is toggled, be called with the mouse event that triggered the dropdown.

I have also made a change to dropdown-toggle so it stops propagation, since this could e.g. trigger an accordion to toggle if the dropdown-toggle was placed inside an accordion header.

### CODE REVIEW

File: src/dropdown/dropdown.js [hunk 1]
```diff
@@ -156,4 +177,54 @@ angular.module('ui.bootstrap.dropdown', [])
...
+      }
+      dropdownCtrl.register(onOpen);
+      function onOpen(coords) {
+        if (!coords) {
+          return;
+        }
+        // Hide it before moving it in to the right place
+        element.css('display', 'none');
+        // Needs to be done in a timeout, otherwise it is not visible and thus has no width/height
+        $timeout(function () {
```
Comment (wesleycho) [inline src/dropdown/dropdown.js]: This should support usage of $animate I think, in case the user wants to animate in the menu

**Comment by wesleycho** [comment]
Tests are failing.

This sounds like an interesting feature, although I'm wondering whether it is one we want to support as a part of the core.

Marking needs review, hopefully we have another comment here.

File: src/dropdown/dropdown.js [hunk 2]
```diff
@@ -156,4 +177,54 @@ angular.module('ui.bootstrap.dropdown', [])
...
+            top = top - (totalHeight - docHeight);
+          }
+          if (totalWidth > docWidth) {
+            left = left - (totalWidth - docWidth);
+          }
+          element.css('position', 'fixed');
+          element.css('top', top + 'px');
+          element.css('left', left + 'px');
+        });
+      }
```
Comment (RobJacobs) [inline src/dropdown/dropdown.js]: Could the position service be used here instead?

File: src/dropdown/docs/demo.js [hunk 3]
```diff
@@ -18,4 +18,20 @@ angular.module('ui.bootstrap.demo').controller('DropdownCtrl', function ($scope)
...
   };
+})
+
+.directive('contextMenuToggle', function () {
+  return {
+    require: '^dropdown',
+    link: function (scope, element, attrs, dropdownCtrl) {
+      element.bind('contextmenu', function (evt) {
+        scope.$apply(function () {
+          console.log('context menu right clicked');
```
Comment (wesleycho) [inline src/dropdown/docs/demo.js]: Remove console.log

File: src/dropdown/dropdown.js [hunk 4]
```diff
@@ -135,6 +155,7 @@ angular.module('ui.bootstrap.dropdown', [])
 
       var toggleDropdown = function(event) {
         event.preventDefault();
+        event.stopPropagation();
```
Comment (wesleycho) [inline src/dropdown/dropdown.js]: This is a breaking change I believe, since there is a listener on `document` for the click event.

**Comment by wesleycho** [comment]
Closing due to age & lack of response.

It should be noted that we intend on fixing position issues in 1.2

**Pull request closed without merging**